### PR TITLE
Bridge for views

### PR DESCRIPTION
**Pull request by byencho**
This PR adds support for saving the state of `View` objects (see https://github.com/livefront/bridge/issues/10). To do so, you can now pass a `ViewSavedStateHandler` to the `Bridge.initialize` method. Calling `Bridge` in the view then looks like : 

```kotlin
    override fun onSaveInstanceState(): Parcelable? {
        return Bridge.saveInstanceState(this, super.onSaveInstanceState())
    }

    override fun onRestoreInstanceState(state: Parcelable?) {
        super.onRestoreInstanceState(Bridge.restoreInstanceState(this, state))
    }
```

In the process, I've cleaned up `BridgeDelegate` a bit with some helper methods. 

Below is the main update I will be making to the README when this feature is released.

----

<a name="views"></a>
## Bridge for Views

In addition to `Activity`, `Fragment`, presenter, etc. classes, `Bridge` can also be used to assist in saving the state of `View` classes:

```java
    @Override
    protected Parcelable onSaveInstanceState() {
        return Bridge.saveInstanceState(this, super.onSaveInstanceState());
    }

    @Override
    protected void onRestoreInstanceState(Parcelable state) {
        super.onRestoreInstanceState(Bridge.restoreInstanceState(this, state));
    }
```

In order to enable this ability, a `ViewSavedStateHandler` must be passed to the `Bridge.initialize` method. For example:

```java
        Bridge.initialize(
                this,
                new SavedStateHandler() {
                    ...
                },
                new ViewSavedStateHandler() {
                    @NonNull
                    @Override
                    public <T extends View> Parcelable saveInstanceState(
                            @NonNull T target,
                            @Nullable Parcelable parentState) {
                        return Icepick.saveInstanceState(target, parentState);
                    }

                    @Nullable
                    @Override
                    public <T extends View> Parcelable restoreInstanceState(
                            @NonNull T target,
                            @Nullable Parcelable state) {
                        return Icepick.restoreInstanceState(target, state);
                    }
                }
        );
```

### CODE REVIEW

File: bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java [hunk 1]
```diff
@@ -105,9 +105,15 @@ private boolean isAppInForeground() {
      * for this task (and any others currently running in the background) to complete before
      * proceeding in order to prevent the app from becoming fully "stopped" (and therefore killable
      * by the OS before the data is saved).
+     *
+     * Note that if there is a configuration change is taking place, no action will be taken.
```
Comment (weisers) [inline bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java]: Remove the second "is"

File: bridgesample/src/main/java/com/livefront/bridgesample/common/view/BitmapGeneratorView.kt [hunk 2]
```diff
@@ -43,7 +47,14 @@ class BitmapGeneratorView @JvmOverloads constructor(
...
     init {
         inflate(context, R.layout.view_bitmap_generator_content, this)
-        navigateButton.setOnClickListener { onNavigateButtonClickListener?.invoke() }
+        navigateButton.setOnClickListener {
+            onNavigateButtonClickListener
+                    ?.invoke()
+                    ?: (context as Activity).startActivity(
+                            Intent(context, SuccessActivity::class.java)
+                    )
+
```
Comment (weisers) [inline bridgesample/src/main/java/com/livefront/bridgesample/common/view/BitmapGeneratorView.kt]: Extra line

File: bridgesample/src/main/java/com/livefront/bridgesample/scenario/view/LargeDataView.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    override fun onSaveInstanceState(): Parcelable? {
+        return Bridge.saveInstanceState(this, super.onSaveInstanceState())
+    }
+
+    override fun onRestoreInstanceState(state: Parcelable?) {
+        super.onRestoreInstanceState(Bridge.restoreInstanceState(this, state))
+        bitmapGeneratorView.generatedBitmap = savedBitmap
+    }
+
```
Comment (weisers) [inline bridgesample/src/main/java/com/livefront/bridgesample/scenario/view/LargeDataView.kt]: Extra line (unless you've been including these for this project)

File: bridgesample/src/main/java/com/livefront/bridgesample/scenario/view/NonBridgeLargeDataView.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    override fun onSaveInstanceState(): Parcelable? {
+        return StateSaver.saveInstanceState(this, super.onSaveInstanceState())
+    }
+
+    override fun onRestoreInstanceState(state: Parcelable?) {
+        super.onRestoreInstanceState(StateSaver.restoreInstanceState(this, state))
+        bitmapGeneratorView.generatedBitmap = savedBitmap
+    }
+
```
Comment (weisers) [inline bridgesample/src/main/java/com/livefront/bridgesample/scenario/view/NonBridgeLargeDataView.kt]: Here too

File: bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java (hunk 1 above)
Comment (byencho) [reply bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java]: Sure.

File: bridgesample/src/main/java/com/livefront/bridgesample/scenario/view/LargeDataView.kt (hunk 3 above)
Comment (byencho) [reply bridgesample/src/main/java/com/livefront/bridgesample/scenario/view/LargeDataView.kt]: No, I think I don't want these. I'll remove them.

File: bridgesample/src/main/java/com/livefront/bridgesample/scenario/view/NonBridgeLargeDataView.kt (hunk 4 above)
Comment (byencho) [reply bridgesample/src/main/java/com/livefront/bridgesample/scenario/view/NonBridgeLargeDataView.kt]: OK, these have all been fixed.

File: bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java (hunk 1 above)
Comment (byencho) [reply bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java]: Done.

**Review by weisers — APPROVED** [review]

**Comment by byencho** [comment]
Thanks @weisers

**Pull request merged**